pkg/updatecheck/updatecheck_test.go

```diff
@@ -0,0 +1,87 @@
+package updatecheck
+
+import (
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	"time"
+
+	"github.com/drud/ddev/pkg/testcommon"
+	"github.com/drud/ddev/pkg/version"
+	"github.com/stretchr/testify/assert"
+)
+
+const testOrg = "drud"
+const testRepo = "ddev"
+
+// TestGetContainerHealth tests the function for processing container readiness.
+func TestUpdateNeeded(t *testing.T) {
+	assert := assert.New(t)
+	tmpdir := testcommon.CreateTmpDir("TestUpdateNeeded")
+	updateFile := filepath.Join(tmpdir, ".update")
+
+	// Ensure updates are required when the update file doesn't exist yet.
+	updateRequired, err := IsUpdateNeeded(updateFile, time.Duration(60)*time.Second)
+	assert.True(updateRequired, "Update is not required when the update file does not exist")
+	assert.NoError(err)
+
+	// Ensure updates are not required when the update duration is impossibly far in the future.
+	updateRequired, err = IsUpdateNeeded(updateFile, time.Duration(9999999)*time.Second)
+	assert.False(updateRequired, "Update is not required when the update file does not exist")
+	assert.NoError(err)
+
+	// Sleep for 2 seconds.
+	time.Sleep(time.Duration(2) * time.Second)
```

Comment:
Could be wrong, but I don't think this does what you want. time.Duration takes ns right?  Do you want time.Sleep(2 * time.Second) ?  Sad to see a sleep but doesn't seem like it's the end of the world.